djhtml/modes.py

```diff
@@ -323,6 +331,20 @@ def create_token(self, raw_token, src):
         kind = "javascript"
         self.next_mode = self
 
+        if raw_token.lstrip().startswith("switch"):
+            self.hard_indent = True
```

Comment:
This is wrong, although it seems to work. It is wrong because modes are persistent and stateful. As soon as you set `self.hard_indent` to True, all subsequent javascript code will generate "hard" tokens. For example:

```javascript
switch(a) {
    case 1:
}

function foo() { // <-- will be an OpenHard token
}
```

Reply:
Solved in [commit](https://github.com/rtts/djhtml/pull/49/commits/0086311c0b6e8aa360d343f34fe5ecf6f66c4d4d)